chainerx_cc/examples/mnist_cc/mnist.cc

```diff
@@ -0,0 +1,76 @@
+#include "mnist.h"
+
+#include <cassert>
+#include <cstdint>
+#include <fstream>
+#include <memory>
+#include <string>
+
+#include "chainerx/array.h"
+#include "chainerx/device.h"
+#include "chainerx/dtype.h"
+#include "chainerx/routines/creation.h"
+#include "chainerx/shape.h"
+
+namespace chx = chainerx;
+
+namespace {
+
+chx::Array ReadArray(std::ifstream& ifs, const chx::Shape& shape) {
+    int64_t n = shape.GetTotalSize();
+
+    std::shared_ptr<uint8_t> data{new uint8_t[n], std::default_delete<uint8_t[]>{}};
+    ifs.read(reinterpret_cast<char*>(data.get()), n);
+
+    return chx::FromContiguousHostData(shape, chx::Dtype::kUInt8, static_cast<std::shared_ptr<void>>(data), chx::GetDefaultDevice());
+}
+
+int32_t ReadInt32(std::ifstream& ifs) {
+    uint32_t result = 0;
+    for (int i = 0; i < 4; ++i) {
+        char byte;
```

Comment:
Always initialize variables.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-always

```suggestion
        char byte{};
```

Reply:
Fixed.